flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -130,95 +130,108 @@ public void checkpointState(
-					checkpointMetrics.getAlignmentDurationNanos() / 1_000_000,
-					checkpointMetrics.getSyncDurationMillis());
-			}
-		} catch (Exception ex) {
-			// Cleanup to release resources
-			for (OperatorSnapshotFutures operatorSnapshotResult : operatorSnapshotsInProgress.values()) {
-				if (null != operatorSnapshotResult) {
-					try {
-						operatorSnapshotResult.cancel();
-					} catch (Exception e) {
-						LOG.warn("Could not properly cancel an operator snapshot result.", e);
-					}
+		for (OperatorSnapshotFutures operatorSnapshotResult : operatorSnapshotsInProgress.values()) {
+			if (operatorSnapshotResult != null) {
+				try {
+					operatorSnapshotResult.cancel();
+				} catch (Exception e) {
+					LOG.warn("Could not properly cancel an operator snapshot result.", e);
 				}
 			}
+		}
 
-			if (LOG.isDebugEnabled()) {
-				LOG.debug(
-					"{} - did NOT finish synchronous part of checkpoint {}. Alignment duration: {} ms, snapshot duration {} ms",
-					taskName, checkpointMetaData.getCheckpointId(),
-					checkpointMetrics.getAlignmentDurationNanos() / 1_000_000,
-					checkpointMetrics.getSyncDurationMillis());
-			}
+		if (LOG.isDebugEnabled()) {
+			LOG.debug(
+				"{} - did NOT finish synchronous part of checkpoint {}. Alignment duration: {} ms, snapshot duration {} ms",
+				taskName, metadata.getCheckpointId(),
+				metrics.getAlignmentDurationNanos() / 1_000_000,
+				metrics.getSyncDurationMillis());
+		}
 
-			if (checkpointOptions.getCheckpointType().isSynchronous()) {
-				// in the case of a synchronous checkpoint, we always rethrow the exception,
-				// so that the task fails.
-				// this is because the intention is always to stop the job after this checkpointing
-				// operation, and without the failure, the task would go back to normal execution.
-				throw ex;
-			} else {
-				env.declineCheckpoint(checkpointMetaData.getCheckpointId(), ex);
-			}
+		if (options.getCheckpointType().isSynchronous()) {
+			// in the case of a synchronous checkpoint, we always rethrow the exception,
+			// so that the task fails.
+			// this is because the intention is always to stop the job after this checkpointing
+			// operation, and without the failure, the task would go back to normal execution.
+			throw ex;
+		} else {
+			env.declineCheckpoint(metadata.getCheckpointId(), ex);
+		}
+	}
+
+	private void finishAndReportAsync(Map<OperatorID, OperatorSnapshotFutures> snapshotFutures, CheckpointMetaData metadata, CheckpointMetrics metrics) {
+		// we are transferring ownership over snapshotInProgressList for cleanup to the thread, active on submit
+		executorService.execute(new AsyncCheckpointRunnable(
+			snapshotFutures,
+			metadata,
+			metrics,
+			System.nanoTime(),
+			taskName,
+			closeableRegistry,
+			env,
+			asyncExceptionHandler));
+	}
+
+	private void takeSnapshotSync(
+		Map<OperatorID, OperatorSnapshotFutures> operatorSnapshotsInProgress, CheckpointMetaData checkpointMetaData,
```

Comment:
it is not suggested putting multiple arguments in one line. split line for every argument